### PR TITLE
IGNITE-20660 SQL: broken behaviour of OCTET_LENGTH function

### PR DESCRIPTION
**Pull request by AMashenkov**
Thank you for submitting the pull request.

To streamline the review process of the patch and ensure better code quality
we ask both an author and a reviewer to verify the following:

### The Review Checklist
- [ ] **Formal criteria:** TC status, codestyle, mandatory documentation. Also make sure to complete the following:  
\- There is a single JIRA ticket related to the pull request.  
\- The web-link to the pull request is attached to the JIRA ticket.  
\- The JIRA ticket has the Patch Available state.  
\- The description of the JIRA ticket explains WHAT was made, WHY and HOW.  
\- The pull request title is treated as the final commit message. The following pattern must be used: IGNITE-XXXX Change summary where XXXX - number of JIRA issue.
- [ ] **Design:** new code conforms with the design principles of the components it is added to.
- [ ] **Patch quality:** patch cannot be split into smaller pieces, its size must be reasonable.
- [ ] **Code quality:** code is clean and readable, necessary developer documentation is added if needed.
- [ ] **Tests code quality:** test set covers positive/negative scenarios, happy/edge cases. Tests are effective in terms of execution time and resources.

### Notes
- [Apache Ignite Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Java+Code+Style+Guide)

### CODE REVIEW

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java [hunk 1]
```diff
@@ -125,7 +125,12 @@ public static String toString(ByteString b) {
 
     /** LENGTH(VARBINARY|VARCHAR). */
     public static int length(Object b) {
-        return b instanceof ByteString ? octetLength((ByteString) b) : charLength((String) b);
+        return b instanceof ByteString ? SqlFunctions.octetLength((ByteString) b) : charLength((String) b);
+    }
+
+    /** OCTET_LENGTH(VARBINARY|VARCHAR). */
+    public static int octetLength(Object s) {
```
Comment (korlov42) [inline modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java:132]: it's better to have overloaded version for every supported type of parameters

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java [hunk 2]
```diff
@@ -93,6 +93,10 @@ public void testLengthExpressionWithDynamicParameter() {
                 .withParams(new byte[0])
                 .returns(0).check();
 
+        checkQuery("SELECT OCTET_LENGTH(?)")
+                .withParams("123")
```
Comment (korlov42) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/datatypes/varbinary/ItVarBinaryExpressionTest.java:97]: this test class dedicated to VarBinary type, not VarChar

File: modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItFunctionsTest.java [hunk 3]
```diff
@@ -68,6 +68,13 @@ public void testLength() {
         assertQuery("SELECT LENGTH(NULL)").returns(NULL_RESULT).check();
     }
 
+    @Test
+    public void testOctetLength() {
+        assertQuery("SELECT OCTET_LENGTH('TEST')").returns(4).check();
```
Comment (korlov42) [inline modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItFunctionsTest.java:73]: let's also add different tests for different languages to make sure OCTET_LENGTH differs from regular LENGTH

File: modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java (hunk 1 above)
Comment (zstan) [reply modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/IgniteSqlFunctions.java:132]: +1 for overload

**Review by zstan — APPROVED** [review]

**Review by korlov42 — APPROVED** [review]

**Pull request merged**